src/level_set/FESurfaceDistanceEvaluator.cpp

```diff
@@ -357,18 +348,229 @@ FESurfaceDistanceEvaluator::computeSignedDistance(int n_idx, int d_idx)
     TBOX_ASSERT(d_idx >= 0);
 #endif
 
-    if (d_use_vol_extracted_bdry_mesh)
+    // Loop over patches on finest level.
+    const int finest_ln = d_patch_hierarchy->getFinestLevelNumber();
+    Pointer<PatchLevel<NDIM> > level = d_patch_hierarchy->getPatchLevel(finest_ln);
+    for (PatchLevel<NDIM>::Iterator p(level); p; p++)
     {
-        computeSignedDistanceVolExtractedBdryMesh(n_idx, d_idx);
+        Pointer<Patch<NDIM> > patch = level->getPatch(p());
+        const Box<NDIM>& patch_box = patch->getBox();
+        const SAMRAI::hier::Index<NDIM>& patch_lower_index = patch_box.lower();
+        Pointer<CartesianPatchGeometry<NDIM> > patch_geom = patch->getPatchGeometry();
+        const double* patch_X_lower = patch_geom->getXLower();
+        const double* const patch_dx = patch_geom->getDx();
+        Pointer<CellData<NDIM, double> > n_data = patch->getPatchData(n_idx);
+        Pointer<CellData<NDIM, double> > d_data = patch->getPatchData(d_idx);
+
+        // Note that we only work with cells that satisfy the intersecting criteria.
+        for (Box<NDIM>::Iterator it(patch_box); it; it++)
+        {
+            CellIndex<NDIM> ci(it());
+            const bool contains_key = (d_cell_elem_neighbor_map.find(ci) != d_cell_elem_neighbor_map.end());
+            if (contains_key)
+            {
+                std::set<Elem*> elem_set = d_cell_elem_neighbor_map[ci];
```

Comment:
This should be a reference
`std::set<Elem*>& elem_set = d_cell_elem_neighbor_map[ci];`